### DISCUSSION TITLE
Type cast missing for int-to-enum in MESSAGE_INIT

### DISCUSSION THREAD

**Discussion by aamirkhan939** [protobuf-c/protobuf-c#601] [Q&A]
I am working on a project where some files are compiled by arm-none-eabi-g++. Compile always fails in expansion of MESSAGE_INIT where 'enum' variables are assigned 'int' value without type-casting.
The following is an example of where it fails in generated pb-c.h files: 
**********
//example_file.pb-c.h
struct Foo
{
  ProtobufCMessage base;
  enum_type value;
}
#define FOO_INIT \
{ PROTOBUF_C_MESSAGE_INIT{&foo_descriptor} \\
  ,0}

My suggestion would be to add a type-cast for enum values and make it look something like:
#define FOO_INIT \
{ PROTOBUF_C_MESSAGE_INIT{&foo_descriptor} \\
  ,(enum_type)0}

**Comment by lipnitsk** [comment]
Do you have a test case or at least a sample .proto file with this issue? Is it platform-specific?
